### ISSUE TITLE
Total Commander ignores specified installation path on subsequent installations

### ISSUE THREAD

**Issue by jberezanski** [dtgm/chocolatey-packages#189]
The installer of Total Commander tries very hard to discover the location it had been previously installed to on this machine. If it finds it, it will ignore the value specified in INSTALL.INF. It is arguably the correct default behavior, esp. when upgrading.

However, if the user specifies installPath in packageParameters and TC had previously been installed on this machine, the value specified by the user will be silently ignored. This might be surprising and frustrating for the user.

The latest Total Commander installer supports [a flag named "IgnoreOldLocation"](http://ghisler.ch/board/viewtopic.php?t=42172), which supposedly turns off the previous location discovery. Unfortunately, that installer comes only in the self-extracting variant.

Here are some ideas on how to handle it in the package:
1) Implement the same discovery logic as the TC installer and inform the user of the discovered previous installation location.
2) If the user did not specify an explicit installPath, do nothing (let the TC installer run its logic).
3a) If the user did specify an explicit installPath, overwrite all locations examined by the TC installer with the path specified by the user.
3b) Alternative: fail the installation with an error, instructing the user to decide which location should be used, and provide a package parameter named e.g. "-ignoreOldLocation".
4) Reimplement the package to use the self-extracting installer which suports IgnoreOldLocation. Set this flag if the user specifies an explicit installPath.

Here is the full list of registry values and files checked by the TC installer:

```
HKLM\SOFTWARE\Ghisler\Total Commander: IniFileName
HKCU\SOFTWARE\Ghisler\Total Commander: IniFileName
%SystemRoot%\wincmd.ini
HKLM\SOFTWARE\Ghisler\Total Commander: InstallDir
HKCU\SOFTWARE\Ghisler\Total Commander: InstallDir
%AppData%\GHISLER\wincmd.ini
```

In the ini files, install path is stored in the `[Configuration]` section under the `InstallDir` key.
I'm not sure about the precedence of ini files versus InstallDir in the registry - it would require experimentation.

**Comment by dtgm** [comment]
Great! Do you have a pull request? :smile: If you want to take a shot at what you think is best I'm all for it.

**Comment by jberezanski** [comment]
> Do you have a pull request?

Heh, not yet ;-) I'm unfortunately out of my time budget for now, just wanted to write my findings down so that they wouldn't get lost.

> If you want to take a shot at what you think is best 

I believe that would be a combination of 1) and 4). Repackaging the TC installer back to the self-extracing form is not an easy task, however - I had attempted it a couple of months ago, but could not make it work using only free, command line tools. (I was able to do it using TC, following instructions posted by Ghisler in a forum thread, but that would obviously pose a bit of a chicken and egg problem...)

> I'm all for it.

Excellent. I will probably get back to it, once I scrounge some free time.
